### PR TITLE
Fix flakiness of regression test in create_index

### PR DESCRIPTION
**Pull request by hpbee**
A test case in create_index.sql became flaky after commit b5aeb102946bedd3118916e84d79894af7e9f926.
Found issue to be related to stats not being available for table in query sometimes.
So added `ANALYZE <table_name>` before running query, so stats are available always.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by ashwinstar — APPROVED** [review]

**Comment by chrishajas** [comment]
Putting these after the INSERT statements for these tables in `circle.sql` and `geometry.sql` is probably better, since these issues could occur for other queries too depending on when autoanalyze happens to run

File: src/test/regress/expected/create_index.out [hunk 1]
```diff
@@ -338,6 +339,7 @@ SELECT * FROM polygon_tbl WHERE f1 ~ '((1,1),(2,2),(2,1))'::polygon
  ((2,0),(2,4),(0,0))
 (1 row)
 
+ANALYZE circle_tbl;
```
Comment (gpopt) [inline src/test/regress/expected/create_index.out:342]: Can we analyze the table right after inserting data into table instead of before running query?

Comment (gpopt) [reply src/test/regress/expected/create_index.out:342]: oops, just found @chrishajas has the same comment.

Comment (hpbee) [reply src/test/regress/expected/create_index.out]: Done

**Review by gpopt — COMMENTED** [review]
Is analyzing on polygon_tbl not needed anymore, and why? I saw it in the first commit.
,
OK, it is explained in the commit message.

**Review by gpopt — APPROVED** [review]
+1

**Review by chrishajas — APPROVED** [review]

**Comment by chrishajas** [comment]
Can you also do the same for `polygon_tbl` while we're here, I see a flake for that too in https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_main/jobs/icw_gporca_jit_rocky8/builds/754

**Comment by hpbee** [comment]
> Can you also do the same for `polygon_tbl` while we're here, I see a flake for that too

I initially added analyze to `polygon_tbl` as I saw it fail once in my local.
But I din't see it happen again. 
Moreover, there is already `analyze` query run in polygon.sql file. So if it fails again, we will have to find a solution for that separately.
I am going to merge this if it's fine.

**Comment by kainwen** [comment]
Hi @hpbee 
I see a new failure of `create_index` after this PR is pushed: https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_main/jobs/icw_gporca_rocky8/builds/752

**Comment by hpbee** [comment]
> Hi @hpbee I see a new failure of `create_index` after this PR is pushed: https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_main/jobs/icw_gporca_rocky8/builds/752

Thank you for info @kainwen. Working on it right away.

**Pull request merged**